src/ansi-c/library/cprover.h

```diff
@@ -26,8 +26,8 @@ extern const void *__CPROVER_deallocated;
 extern const void *__CPROVER_memory_leak;
 
 extern int __CPROVER_malloc_failure_mode;
-extern __CPROVER_size_t __CPROVER_max_malloc_size;
 extern __CPROVER_bool __CPROVER_malloc_may_fail;
+extern __CPROVER_thread_local __CPROVER_size_t __CPROVER_max_malloc_size;
```

Comment:
Why is this thread local?

Reply:
This is to avoid our concurrency encoding picking the variable as one that can't be constant-propagated.

Reply:
I have now added a comment to clarify this.

Reply:
Would making it `const` achieve that?

Reply:
We don't commonly trust `const` at this time. We could, however, consider a lightweight analysis that establishes that a variable (irrespective of whether it is marked `const` or not) is only ever written once (and doesn't have its address taken).